### ISSUE TITLE
Add focus state for button.

### ISSUE THREAD

**Issue by lf-collins** [vmware/clarity#4400]
## Describe the bug

Clarity button does not have a visual indication of keyboard focus. Keyboard-only users will have difficulty navigating to and interacting with controls that do not convey a visual indication of keyboard focus.

1.  Go to 'https://clarity.design/documentation/buttons'
2.  Use the Tab keyboard to move to different buttons.
3.  Check the button status, outline and flat buttons have a outline style "outline: 1px dotted", but the solid buttons have no state change.

## Expected behavior

Ensure that a well-defined visual focus indicator is applied to buttons and the focus indicator has a minimum color contrast of 3.0:1 with surrounding content to ensure visibility for users with low vision or color blindness.
Related info: https://github.com/vmware/clarity/pull/2929, we provide focus status for button group, but it only works for checkbox button group.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [3.0]

**Device:**

* Type: [MacBook]
* OS: [e.g. iOS]
* Browser [Firefox and IE]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

**Comment by gnomeontherun** [comment]
Hello, I'm having trouble seeing what you are seeing. The buttons have the default browser focus styles. Can you please verify with more details?

**Comment by lf-collins** [comment]
Hi gnomeontherun,

I can reproduce it in Firefox, the button focus status is:
<img width="908" alt="Screen Shot 2020-03-27 at 2 05 35 PM" src="https://user-images.githubusercontent.com/26111301/77727566-94c90100-7035-11ea-829d-5249183a4a87.png">
<img width="893" alt="Screen Shot 2020-03-27 at 2 05 46 PM" src="https://user-images.githubusercontent.com/26111301/77727575-985c8800-7035-11ea-8bae-9382699f1136.png">
<img width="896" alt="Screen Shot 2020-03-27 at 2 05 57 PM" src="https://user-images.githubusercontent.com/26111301/77727582-9a264b80-7035-11ea-894c-e08253758478.png"> The solid button doesn't have a border in Firefox when focusing.

In Chrome, all buttons have a border when focusing, making it easy to distinguish:
<img width="893" alt="Screen Shot 2020-03-27 at 2 06 30 PM" src="https://user-images.githubusercontent.com/26111301/77727646-bc1fce00-7035-11ea-87d5-f1a48cb1a2cd.png">

In IE, it is a dotted border:
![Screen Shot 2020-03-27 at 2 12 05 PM](https://user-images.githubusercontent.com/26111301/77727749-f0938a00-7035-11ea-8229-3dee144d1781.png)
![Screen Shot 2020-03-27 at 2 12 16 PM](https://user-images.githubusercontent.com/26111301/77727753-f25d4d80-7035-11ea-9f61-cdc815281e37.png)
![Screen Shot 2020-03-27 at 2 12 28 PM](https://user-images.githubusercontent.com/26111301/77727759-f5583e00-7035-11ea-9c8d-c89521d0f988.png)

It would be nice to keep the focus style consistent, and the focus style should be easy to distinguish as in Chrome, not a dotted border of 1px.

**Comment by gnomeontherun** [comment]
I think this needs to be evaluated with the upcoming WCAG 2.2 guidelines before we settle on a direction here. Marking for review.

https://www.w3.org/TR/WCAG22/#focus-visible-enhanced

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
